src/libraries/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/Activity.cs

```diff
@@ -1484,6 +1492,21 @@ public void Add(KeyValuePair<string, object?> value)
                 }
             }
 
+            public object? Get(string key)
+            {
+                LinkedListNode<KeyValuePair<string, object?>>? current = _first;
```

Comment:
It would be worth a comment here and in Add/Remove that Get races with Add/Remove because there is no lock here. I think it is a good implementation, it just might catch someone by surprise if they aren't paying attention. For example someone might think it was safe to mutate the node after it had been removed from the list, but that would be a bug if Get() is manipulating the node at the same time.